nighthawk/source/client/benchmark_client_impl.cc

```diff
@@ -124,11 +151,15 @@ void BenchmarkClientHttpImpl::initialize(Envoy::Runtime::Loader& runtime) {
   Envoy::Network::ConnectionSocket::OptionsSharedPtr options =
       std::make_shared<Envoy::Network::ConnectionSocket::Options>();
   if (use_h2_) {
-    pool_ = std::make_unique<Envoy::Http::Http2::ProdConnPoolImpl>(
-        dispatcher_, host, Envoy::Upstream::ResourcePriority::Default, options);
+    pool_ = std::make_unique<H2Pool>(dispatcher_, host, Envoy::Upstream::ResourcePriority::Default,
+                                     options);
   } else {
-    pool_ = std::make_unique<Envoy::Http::Http1::ProdConnPoolImpl>(
-        dispatcher_, host, Envoy::Upstream::ResourcePriority::Default, options);
+    pool_ = std::make_unique<H1Pool>(dispatcher_, host, Envoy::Upstream::ResourcePriority::Default,
+                                     options);
+  }
+
+  if (prefetch_connections_) {
+    prefetchPoolConnections();
```

Comment:
IIUC, what this does is allow you to prefetch at the start of the benchmark. This seems useful. Is there any value on continuing to also prefetch when connections are dropped (due to exceeding max requests)?

Reply:
I think that doing that would add value, maybe as a separate flag; we would make sure we mostly have the same number of connections open throughout the test. As the number of open connections is something that may influence server performance all by itself, it might be interesting.

Reply:
Ack, maybe put it on the backlog and we can resolve.

Reply:
Added to the backlog